python_ta/contracts/__init__.py

```diff
@@ -260,8 +260,18 @@ def _check_function_contracts(wrapped, instance, args, kwargs):
 
 
 def check_type_strict(argname: str, value: Any, expected_type: type) -> None:
-    """Ensure that ``value`` matches ``expected_type``. Differentiates between types float and int."""
-    if type(value) is int and expected_type is float:
+    """Ensure that ``value`` matches ``expected_type``.
+
+    Differentiates between:
+        - float vs. int
+        - bool vs. int
+    """
+    if (
+        type(value) is int
```

Comment:
This condition broken up into multiple lines is a bit hard to read, especially since the precedence rules of `and` vs. `or` aren't well-known.

I would rewrite this condition as

```
if ((type(value) is int and expected_type is float) or
    (type(value) is bool and expected_type is int):
    ...
```

and then let the `black` pre-commit hook format the code from there.